File: src/main/java/com/starpoint/mixins/Trim.java
package com.starpoint.mixins;

import org.apache.tapestry5.Field;
import org.apache.tapestry5.annotations.Import;
import org.apache.tapestry5.annotations.InjectContainer;
import org.apache.tapestry5.ioc.annotations.Inject;
import org.apache.tapestry5.json.JSONObject;
import org.apache.tapestry5.services.javascript.JavaScriptSupport;

/**
 */
@Import(library = "trim.js")
public class Trim {

    @InjectContainer
    private Field component;

    @Inject
    private JavaScriptSupport javaScriptSupport;

    void afterRender() {
        JSONObject specs = new JSONObject("id", component.getClientId());
        javaScriptSupport.addInitializerCall("trimValue", specs);
    }

}

File: src/main/resources/com/starpoint/mixins/trim.js
(function ($) {
    T5.extendInitializers(function() {

        function init(specs) {
            $("#" + specs.id).blur(function () {
                    if (this.value) {
                        this.value = $.trim(this.value);
                    }
                }
            );
        }

        return {
            trimValue : init
        };
    });

})(jQuery);